calculator.rb

```diff
@@ -0,0 +1,89 @@
+# calculator program allows user to perform simple operations (+, -, *, /, ^, %) between two operands
+
+# declare methods for each operation
+def add(num1, num2)
+  return "#{num1} + #{num2} = #{num1 + num2}"
+end
+
+def sub(num1, num2)
+  return "#{num1} - #{num2} = #{num1 - num2}"
+end
+
+def mult(num1, num2)
+  return "#{num1} * #{num2} = #{num1 * num2}"
+end
+
+def div(num1, num2)
+  if num2 == 0
+    puts "Please do not divide by zero. You're going to rip a hole in the time-space continuum!"
+  else
+    return "#{num1} / #{num2} = #{num1 / num2}"
+  end
+end
+
+def exp(num1, num2)
+  expanded_form = ""
+  (1...num2).each do
+    expanded_form += "#{num1} * "
+  end
+  expanded_form += "#{num1}"
+  return "#{num1} ^ #{num2} = #{expanded_form} = #{num1 ** num2}"
+end
+
+def mod(num1, num2)
+  return "#{num1} % #{num2} = #{num1 % num2}"
+end
+
+# validate number
+def val(num)
+  # this should cover strings that don't evaluate to numbers as well as nil
+  while num.to_i == 0 || num.to_f == 0.0
```

Comment:
I like that you break this out into its own function. However, I'm not so sure about the name - `val` doesn't tell me much about what is going on. This means I need to do a little more thinking if I'm reading code where this function is called: is it `value` or `validate`? What am I validating? Spelling it out is a little more typing, but produces much more readable code.